app/models/compute_resource.rb

```diff
@@ -407,6 +407,7 @@ def memory_gb_to_bytes(memory_size)
   end
 
   def to_bool(value)
+    Foreman::Deprecation.deprecation_warning("3.6", "The method to_bool will be repaced by rails implementation.")
```

Comment:
Idea: Maybe it's worth to keep the `to_bool` method and just change the code to:
```ruby
def to_bool(value)
  ActiveRecord::Type::Boolean.new.deserialize(value)
end
```

With this plugin maintainer won't have to update their code base and we still have consistent behavior in the core.

Reply:
Well, most of the time we would do that, but as @ezr-ondrej mentioned, Foreman's `to_bool` behave a bit differently from Rails' `deserialize`, so if we simply change the method definition we potentially could break stuff for plugins :/

Reply:
Yeah I guess you are right, if I'm thinking about that this would require extensive testing of all compute resources & plugins, that's not worth the hassle. 

@Dyrkon Let's revert this change back and keep the original implementation. Let's just focus only on `Foreman::Cast.to_bool`

Reply:
Yes, that's better approach.